source/_data/releases.cjs

```diff
@@ -0,0 +1,90 @@
+const { spawn: nodeSpawn } = require('node:child_process');
+const fs = require('node:fs/promises');
+
+const VERSION_CACHE_PATH = './source/_data/versionCache.json';
+
+// Promise version of `spawn` to avoid blocking the main thread while waiting
+// for the child processes
+function spawn(cmd, args, options) {
+  return new Promise((resolve, reject) => {
+    const child = nodeSpawn(cmd, args, options);
+    const stderr = [];
+    const stdout = [];
+    child.stdout.on('data', (data) => {
+      stdout.push(data.toString());
+    });
+    child.on('error', (e) => {
+      stderr.push(e.toString());
+    });
+    child.on('close', () => {
+      if (stderr.length) reject(stderr.join(''));
+      else resolve(stdout.join(''));
+    });
+  });
+}
+
+// Retrieve the highest stable version of `repo`, based on its git tags
+// Store results in `VERSION_CACHE_PATH` for faster future runs
+async function getLatestVersion(repo) {
+  let versionCache;
+  try {
+    versionCache = JSON.parse(await fs.readFile(VERSION_CACHE_PATH));
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      versionCache = {}; // Cache is missing and needs to be created
+    } else {
+      throw err;
+    }
+  }
+  let latestVersion = versionCache[repo];
+  if (latestVersion !== undefined) return latestVersion;
```

Comment:
After the first run, the cache should be present and no network requests are done

Reply:
I realize in production the file shouldn't exist, but now (https://github.com/oddbird/sass-site/commit/5721ecc5bb74031630d3d6992bb4c6ff74bc3063) locally a `build` will regenerate the cache. If we want that on every _initial_ build (even during `serve`), we could move the cache into memory instead of the fs.